airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/build_image/steps/manifest_only_connectors.py

```diff
@@ -0,0 +1,67 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+
+from typing import Any
+
+from dagger import Container, Platform
+from pipelines.airbyte_ci.connectors.build_image.steps import build_customization
+from pipelines.airbyte_ci.connectors.build_image.steps.common import BuildConnectorImagesBase
+from pipelines.airbyte_ci.connectors.context import ConnectorContext
+from pipelines.dagger.actions.python.common import with_python_connector_installed
+from pipelines.models.steps import StepResult
+from pydash.objects import get
+
+
+class BuildConnectorImages(BuildConnectorImagesBase):
+    """
+    A step to build a manifest only connector image.
+    A spec command is run on the container to validate it was built successfully.
+    """
+
+    context: ConnectorContext
+    PATH_TO_INTEGRATION_CODE = "/airbyte/integration_code"
+
+    async def _build_connector(self, platform: Platform, *args: Any) -> Container:
+        baseImage = get(self.context.connector.metadata, "connectorBuildOptions.baseImage")
+        if not baseImage:
+            raise ValueError("connectorBuildOptions.baseImage is required to build a manifest only connector.")
+
+        return await self._build_from_base_image(platform)
+
+    def _get_base_container(self, platform: Platform) -> Container:
+        base_image_name = get(self.context.connector.metadata, "connectorBuildOptions.baseImage")
+        self.logger.info(f"Building manifest connector from base image {base_image_name}")
+        return self.dagger_client.container(platform=platform).from_(base_image_name)
+
+    async def _build_from_base_image(self, platform: Platform) -> Container:
+        """Build the connector container using the base image defined in the metadata, in the connectorBuildOptions.baseImage field.
+
+        Returns:
+            Container: The connector container built from the base image.
+        """
+        self.logger.info(f"Building connector from base image in metadata for {platform}")
+        base = self._get_base_container(platform)
+
+        customized_base = await build_customization.pre_install_hooks(self.context.connector, base, self.logger)
```

Comment:
I think there's duplication between `_build_from_base_image` in this class vs the python one. Can we keep a common code path for setting entrypoints, labels, calling pre/post hooks etc?

Your could might "just be" considered a  pre-install hook?

Reply:
@alafanechere I was thinking on this. 

Im not sure this is the place to DRY up the code **completely** and have a common build for python and manifest only connectors.

The reasons for that are
1. The similarities between the manifest and python `_build_from_base_image` are that they use pre and post commit hooks. After that the python build does a lot more inbetween with two stage build
2. We are likely going to deviate more from python as we look at how we add support for custom components, potentially in other languages.

I'm almost more inclined to 
1. Remove the hooks for manifest only
2. Add a helper function for adding the docker labels that the two builds can share